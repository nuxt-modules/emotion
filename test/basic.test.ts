import { fileURLToPath } from 'node:url'
import { describe, it, expect } from 'vitest'
import { setup, $fetch } from '@nuxt/test-utils'
describe('nust emotion ssr', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('../playground', import.meta.url))
  })
  it('renders the index page', async () => {
    const html = await $fetch('/')
    expect(html).toContain('<style data-emotion')
    expect(html).toContain('<script data-emotion')
    expect(html).toContain('window.$emotionIds')

    // Removed because the nuxt build entries product difference SSR output
    // for each test run so it doesn't make sense to run snapshot tests.
    // expect(html).toMatchSnapshot()
  })
})
