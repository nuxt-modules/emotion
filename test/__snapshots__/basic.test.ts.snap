// Vitest Snapshot v1

exports[`nust emotion ssr > renders the index page 1`] = `
"<!DOCTYPE html>
<html >
<head><meta charset=\\"utf-8\\">
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"><link rel=\\"modulepreload\\" as=\\"script\\" crossorigin href=\\"/_nuxt/entry.52d2c064.js\\"><link rel=\\"prefetch\\" as=\\"script\\" crossorigin href=\\"/_nuxt/error-component.46dc171f.js\\"><style data-emotion=\\"\\"></style><script data-emotion=\\"\\">window.$emotionIds=[]</script></head>
<body ><div id=\\"__nuxt\\"><div><div class=\\"css-camw5v\\"> Nuxt module playground! </div><button class=\\"css-u6jiew\\"> Hello World </button></div></div><script>window.__NUXT__={data:{},state:{},_errors:{},serverRendered:true,config:{public:{},app:{baseURL:\\"\\\\u002F\\",buildAssetsDir:\\"\\\\u002F_nuxt\\\\u002F\\",cdnURL:\\"\\"}}}</script><script type=\\"module\\" src=\\"/_nuxt/entry.52d2c064.js\\" crossorigin></script></body>
</html>"
`;
